src/client/diagnostics.rs

```diff
@@ -0,0 +1,137 @@
+use bevy::diagnostic::DiagnosticId;
+use bevy::{
+    diagnostic::{Diagnostic, Diagnostics, RegisterDiagnostic},
+    prelude::*,
+    time::common_conditions::on_timer,
+};
+use std::time::Duration;
+
+/// Collects replication stats during packet processing
+/// flushed to Diagnostics system periodically.
+#[derive(Default, Resource, Debug)]
+pub struct ReplicationStats {
+    // incremented per entity that changes
+    pub entities_changed: u32,
+    // incremented for every component that changes
+    pub components_changed: u32,
+    // incremented per client mapping added
+    pub mappings: u32,
+    // incrementer per entity despawn
+    pub despawns: u32,
+    // replication packets recvd
+    pub packets: u32,
+    // replication bytes received as packet payload (not transport layer headers, etc)
+    pub bytes: u32,
+}
+
+pub mod replication_diagnostics {
+    use super::*;
+    /// How many entities modified per second by replication
+    pub const REPLICATED_ENTITY_CHANGES_DIAGNOSTIC: DiagnosticId =
```

Comment:
Let's remove `_DIAGNOTICS` suffix and maybe `REPLICATED_` prefix. Also I would remove comments too, it's obvious from the name. See how it's done in Bevy: https://github.com/bevyengine/bevy/blob/main/crates/bevy_diagnostic/src/frame_time_diagnostics_plugin.rs
```suggestion
    pub const ENTITY_CHANGES: DiagnosticId =
```